go/arrow/example_test.go

```diff
@@ -593,3 +593,66 @@ func Example_table() {
 	// rec[3]["f1-i32"]: [16 17 18 19 20]
 	// rec[3]["f2-f64"]: [16 17 18 19 20]
 }
+
+// This example demonstrates how to create a Map Array.
+// The resulting array should be:
```

Comment:
should the comment at the end of the method be moved up here?  Maybe provide a reference here instead of documentation of usage I mentioned above.

Reply:
By putting the comment at the end of the method here in that format this example actually gets run as a test when running tests and confirms that the output of running this method matches the output comment at the end of the method. moving the comment at the end of the method would disable that benefit.

Reply:
ahh, very cool.

Reply:
The other cool aspect of this is that it shows up in the go docs too as an example like the examples on the pkg.go.dev docs [here](https://pkg.go.dev/github.com/apache/arrow/go/arrow#example-package-FixedSizeListArray) 😄